clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1294,7 +1309,10 @@ public void handleResponse(AbstractResponse response) {
                 reenqueue();
             } else if (error == Errors.TRANSACTIONAL_ID_AUTHORIZATION_FAILED ||
                     error == Errors.CLUSTER_AUTHORIZATION_FAILED) {
-                fatalError(error.exception());
+                log.info("Abortable authorization error: {}.  Transition the producer state to {}", error.message(), State.ABORTABLE_ERROR);
+                lastError = error.exception();
+                epochBumpRequired = true;
```

Comment:
bumping epoch here - I think sender should bump it on the next iteration.

Reply:
can we confirm this works as intended via test?

Reply:
hey sorry - i'm in progress of writing a test here 😅

Reply:
Was about to ask about this and realized we removed the bump. I guess we could still have a test unless we think the other testing covers this end to end.

Reply:
even i think the initProducerId path in the sender has been tested, but it's probably worth adding a sender test to make sure the epoch is bumped from -1 to 0. I think it should be an easy modification, 👀

Reply:
@jolshan  - See the last commit.  I cleaned up the comments because authorization isn't a fatal error after the patch.  I also patched a test to demonstrate resend after fixing the permission should bring the producer back to normal. Also tested the epoch there to ensure it is 0, not -1 after fixing the permission.